R/g.part5_analyseSegment.R

```diff
@@ -47,7 +47,7 @@ g.part5_analyseSegment = function(indexlog, timeList, levelList,
   #==========================
   # The following is to avoid issue with merging sleep variables from part 4
   # Note that this means that for MM windows there can be multiple or no wake or onsets in window
-  date = as.Date(ts$time[segStart], tz = params_general[["desiredtz"]])
+  date = as.Date(ts$time[qqq[1]], tz = params_general[["desiredtz"]]) # changed segStart for qqq[1] in case segment is not available in this day
```

Comment:
Does this imply that the date in the output will no longer be the date on which the segment started but the date on which the window started? If yes, then that may not be intuitive.
How about inserting `ts$time[ifelse(is.na(segStart), yes = qqq[1], no =segStart])]` ?

Reply:
Oh yes, that is true.

Reply:
This discussion made me realize that we can take advantage of this to simplify the identification of the segment limits. As the date is known, we can format the `qwindow` as `"%H:%M:%S"` and then get first and last epoch within the defined limits. This simplifies the code and makes it less dependent on epoch size and on whether the segment is fully or partially available in ts.

The last commit contains this proposal.